src/inverted-index.js

```diff
@@ -0,0 +1,272 @@
+class Index {
+
+  /**
+   * [constructor Creates Object variables to hold 
+   * 1. [jsonDatabase] Uploaded files
+   * 2. [indexFile] the indexed filenames and their contents 
+   * 3. [searchResult] the search results
+   * @return {[type]} [description]
+   */
+  constructor() {
+    this.jsonDatabase = {};
+    this.indexFile = {};
+    this.searchResult;
+  }
+
+  /**
+   * [saveUploads creates a key and value object item that stores the uploaded file(s)]
+   * @param  {[string]} fileName [filename]
+   * @param  {[object]} jsonFile [content of uploaded json file]
+   * @return {[boolean]}          [returns true on succesful addition of object to datatbase]
+   */
+  saveUploads(fileName, jsonFile) {
+      if (!this.isValid(fileName, jsonFile)) {
+        return false;
+      }
+      if (typeof jsonFile === 'string') {
+        jsonFile = JSON.parse(jsonFile);
+      }
+
+      this.jsonDatabase[fileName] = [];
+      for (const docObject in jsonFile) {
```

Comment:
**Code Quality - Nitpick**
Use `let` to declare iterator variables as the are always reassigned.